fud2/fud-core/src/cli.rs

```diff
@@ -237,7 +239,23 @@ fn get_resource(driver: &Driver, cmd: GetResource) -> anyhow::Result<()> {
 }
 
 pub fn cli(driver: &Driver) -> anyhow::Result<()> {
-    let args: FakeArgs = argh::from_env();
+    let mut args: FakeArgs = argh::from_env();
+
+    // inv: `None` unless neither `--keep` nor `--dir` were passed AND no
+    // subcommand is passed, in which case it is `Some(temp_dir)`: guaranteed
+    // to be unique and dropped at the end of the process
+    let mut _unique_build_dir = None;
+    if args.keep.is_none() && args.dir.is_none() && args.sub.is_none() {
+        let temp_dir = TempDir::new_in(".", ".fud-")?;
```

Comment:
I don't think the `tempdir` create is necessary here. Namely, it does more than we need it to do: it not only generates a name but also creates and deletes the directory. We already have machinery to create and (conditionally) delete the directory, here:
https://github.com/calyxir/calyx/blob/513fda8bff119eea97e645b19dbbacf674f8a5a0/fud2/fud-core/src/run.rs#L206

To keep things as simple as possible, I think we should have only one place where we create the directory and delete the directory. So here (or perhaps in `emit_and_run` or `emit_to_dir`, based on a new option that says "generate a unique directory name"), we can simply generate the name and reuse the existing machinery for managing the directory's creation/cleanup. This could even be changed to use `tempdir` if we want! But I think there must be only one implementation of the creation/deletion logic, not two.

Reply:
I'm not familiar with the `argh` crate, so I don't know if we could embed a `tempdir::TempDir` in the argument struct while also allowing for the option to specify a directory. Were you thinking of an entirely new option for "use a unique temporary directory" instead of taking the absence of `--keep` and `--dir` as initation?